### PR TITLE
mkpkg: Grab sdist instead of first URL from PyPI.

### PR DESCRIPTION
**Pull request by msabramo**
The first URL from PyPI is often a platform-specific wheel, so it's better if we prefer to grab a source distribution that we can build with emscripten.

Before:

```
$ bin/pyodide mkpkg lxml
$ grep 'url: ' packages/lxml/meta.yaml
  url: https://files.pythonhosted.org/packages/28/32/951337aa3d14e58cae3f585e468e402141f5ac3728589eccca765c4e17ce/lxml-4.3.2-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
```

After:

```
$ bin/pyodide mkpkg lxml
$ grep 'url: ' packages/lxml/meta.yaml
  url: https://files.pythonhosted.org/packages/65/6f/d070609b415f4a24bf36b7deb94bfd791e8d023c0dbf4233659af53fe0ab/lxml-4.3.2.tar.gz
```

Closes: #363.

### CODE REVIEW

File: pyodide_build/mkpkg.py [hunk 1]
```diff
@@ -10,6 +10,12 @@
 PACKAGES_ROOT = Path(__file__).parent.parent / 'packages'
 
 
+def get_sdist_url_entry(json_content):
+    for entry in json_content['urls']:
+        if entry['filename'].endswith('.tar.gz'):
+            return entry
+
```
Comment (mdboom) [inline pyodide_build/mkpkg.py]: We should raise an exception here if it didn't find anything suitable.

File: pyodide_build/mkpkg.py [hunk 2]
```diff
@@ -10,6 +10,12 @@
 PACKAGES_ROOT = Path(__file__).parent.parent / 'packages'
 
 
+def get_sdist_url_entry(json_content):
+    for entry in json_content['urls']:
+        if entry['filename'].endswith('.tar.gz'):
```
Comment (mdboom) [inline pyodide_build/mkpkg.py]: It actually could be any of the formats supported by [shutil.unpack_archive](file:///usr/share/doc/python3-docs/html/library/shutil.html?highlight=unpack_archive#shutil.unpack_archive)...  but maybe this is the most common and good enough for now.

File: pyodide_build/mkpkg.py (hunk 1 above)
Comment (msabramo) [reply pyodide_build/mkpkg.py]: OK, I force-pushed a new version that raises an exception.

File: pyodide_build/mkpkg.py (hunk 2 above)
Comment (msabramo) [reply pyodide_build/mkpkg.py]: OK, I force-pushed a new version that looks at all of the different extensions returned by `shutil.get_unpack_formats()`.

**Comment by msabramo** [comment]
@mdboom: This look good to you?

**Pull request merged**